src/ape/api/providers.py

```diff
@@ -379,6 +379,16 @@ def mine(self, num_blocks: int = 1):
             NotImplementedError: Unless overridden.
         """
 
+    @raises_not_implemented
+    def set_balance(self, address: AddressType, amount: int):
```

Comment:
Thank you for discussing this choice and future alternatives to making these test methods, like setting the balance or timestamp, only available on local/test chains.